src/main/java/org/webshar/hrms/controller/ResourceController.java

```diff
@@ -0,0 +1,117 @@
+package org.webshar.hrms.controller;
+
+import java.util.Collections;
+import java.util.List;
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+import org.apache.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.webshar.hrms.model.db.Resource;
+import org.webshar.hrms.request.resource.ResourceCreateRequest;
+import org.webshar.hrms.request.resource.ResourceUpdateRequest;
+import org.webshar.hrms.response.BatchResponse;
+import org.webshar.hrms.response.Response;
+import org.webshar.hrms.service.ResourceService;
+import org.webshar.hrms.service.exception.EntityAlreadyExistsException;
+import org.webshar.hrms.service.exception.EntityNotFoundException;
+import org.webshar.hrms.service.exception.ServiceException;
+
+@RestController
+@RequestMapping("/")
+public class ResourceController
+{
+
+  @Autowired
+  ResourceService resourceService;
+
+  private static final Logger LOGGER = Logger.getLogger(ResourceController.class);
+
```

Comment:
Let's use default logging APIs provided by spring boot viz. SLF4J logging APIs. 

(Why to use slf4j APIs: The Simple Logging Facade for Java (SLF4J) serves as a simple facade or abstraction for various logging frameworks, such as java.util.logging, logback and log4j. SLF4J allows the end-user to plug in the desired logging framework at deployment time)

e.g. 
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
private static final Logger LOGGER = LoggerFactory.getLogger(<ClassName>.class);